src/PostgREST/Config/Database.hs

```diff
@@ -181,6 +184,43 @@ queryRoleSettings pgVer prepared =
     rows :: HD.Result [(Text, Maybe Text, [(Text, Text)])]
     rows = HD.rowList $ (,,) <$> column HD.text <*> nullableColumn HD.text <*> compositeArrayColumn ((,) <$> compositeField HD.text <*> compositeField HD.text)
 
+
+queryFuncSettings :: PgVersion -> Bool -> Session FunctionSettings
+queryFuncSettings pgVer prepared =
```

Comment:
Perhaps the query can be inside the SchemaCache? Since this is not meant to be exposed to external users, we could remove the `funcSettings` attribute here:

https://github.com/PostgREST/postgrest/blob/6c3d7a946daef91014328bc50c0f63744fada8ad/src/PostgREST/SchemaCache.hs#L187-L189

Reply:
Then `Routine` can have `funcSettings` (or routSettings) as an additional attribute.